src/api_manager/service_control/aggregated.cc

```diff
@@ -371,8 +371,9 @@ void Aggregated::Check(
           *response, service_control_proto_.service_name(), &response_info);
       on_done(status, response_info);
     } else {
-      // If network_fail_open is true, it is always OK to proceed.
-      if (network_fail_open_) {
+      // Only Code::UNAVAILABLE error is network failure.
```

Comment:
Can you clarify where `Unavailable` is set for non 4xx responses? And what is set for 4xx responses?

Is it here? https://github.com/cloudendpoints/esp/blob/28c924749263c0a8fe252ed3f2c678cb3988534a/src/nginx/http.cc#L993

Reply:
1) [here](https://github.com/cloudendpoints/esp/blob/master/src/nginx/http.cc#L573)  Http status code is copied to Status()

2) [here](https://github.com/cloudendpoints/esp/blob/master/src/api_manager/service_control/aggregated.cc#L607), we convert http status to RpcStatus::code in ToProto() call.  This code is also modifed in this PR.